pkg/nvmeclient/nvme_client.go

```diff
@@ -526,13 +526,14 @@ func ConnectAllNVMEDevices(logPageEntries []*hostapi.NvmeDiscPageEntry, hostnqn
 					// we can't deduce that if the DS is down on that node we will fail to connect cause there might be a network partition
 					// on the discovery-service or the DS is down on that node but the IO controller is still accessible.
 					logrus.WithError(perr).Warnf("failed to connect IO controller. In case we have a node down, discovery provides log-page with it's connect info. we will probably fail to connect to it.")
+					return nil, perr
```

Comment:
Do we really want to bail out here early without trying to connect to all the remaining nodes?
We could instead accumulate the error and return after attempting to connect to all nodes.

Reply:
either way after returning an error you will need to reconnect to all nodes, unless we want to optimize it and reconnect only to the nodes we failed to connect

Reply:
@guy-lb who is "you" here? Is the intent that the caller of `ConnectAllNVMEDevices` will call it again in a loop, and that is what you are triggering below? If so, what will be the effect if some server has gone away forever -- will we be notified about it and remove it from the connection map? Consider whether there needs to be "try for a while, ideally using exponential backoff to decide when to try again, and give up forever at some point" type of behavior here.

Reply:
@muliby-lb yes I meant to the caller of `ConnectAllNVMEDevices` which is a go routing in the `start` of the service.
Once this function (`ConnectAllNVMEDevices`) returns an error, it will go and try to `reconnectToCluster`.
in case a server was disconnected, the connection map will be updated.
regarding "using exponential backoff" - we implement this mechanism in `reconnectToCluster`, I think it's possible to add it here as well

Reply:
If any failure in `ConnectAllNVMEDevices` causes us to get to `reconnectToCluster` and that has retries, I think we are good; we don't need multiple levels of retry, `Connect` can stay single shot as long as the above holds.